File: 02-basic-challenges-2/03-display-likes/display_likes.rb
def display_likes(arr)
  return "no one likes this" if arr.empty?

  if arr.length == 1
    "#{arr[0]} likes this"
  elsif arr.length == 2
    "#{arr[0]} and #{arr[1]} like this"
  elsif arr.length == 3
    "#{arr[0]}, #{arr[1]} and #{arr[2]} like this"
  else
    "#{arr[0]}, #{arr[1]} and #{arr.length - 2} others like this"
  end
end

puts display_likes([])
puts display_likes(["Peter"])
puts display_likes(["Jacob", "Alex"])
puts display_likes(["Max", "John", "Mark"])
puts display_likes(["Alex", "Jacob", "Mark", "Max", "Jill"])

File: 02-basic-challenges-2/03-display-likes/display-likes.js
function displayLikes(namesArray) {
  if (namesArray.length === 0) {
    return "no one likes this";
  } else if (namesArray.length === 1) {
    return `${namesArray[0]} likes this`;
  } else if (namesArray.length === 2) {
    return `${namesArray[0]} and ${namesArray[1]} like this`;
  } else if (namesArray.length === 3) {
    return `${namesArray[0]}, ${namesArray[1]} and ${namesArray[2]} like this`;
  } else {
    return `${namesArray[0]}, ${namesArray[1]} and ${namesArray.length - 2} others like this`;
  }
}

module.exports = displayLikes;
